eta/core/image.py

```diff
@@ -174,17 +174,25 @@ def merge_labels(self, image_labels):
         self.add_image_attributes(image_labels.attrs)
         self.add_objects(image_labels.objects)
 
-    def filter_by_schema(self, schema, remove_objects_without_attrs=False):
+    def filter_by_schema(self, schema):
         '''Removes objects/attributes from this object that are not compliant
         with the given schema.
 
         Args:
             schema: an ImageLabelsSchema
-            remove_objects_without_attrs: (bool) If True, remove objects that no
-                longer have attributes after filtering
         '''
         self.attrs.filter_by_schema(schema.attrs)
-        self.objects.filter_by_schema(schema, remove_objects_without_attrs)
+        self.objects.filter_by_schema(schema)
+
+    def remove_objects_without_attrs(self, labels=None):
+        '''Removes DetectedObjects from this instance that do not have
+        attributes.
+
+        Args:
+            labels: an optional list of object labels to which to restrict
+                attention when filtering. By default, all objects are processed
+        '''
+        self.objects.remove_objects_without_attrs(labels=labels)
```

Comment:
Note that I used the `labels=labels` syntax to invoke the `remove_objects_without_attrs` method.

When invoking methods with keyword arguments, please always use the `name=value` syntax. That way, if someone else comes along and adds a new keyword argument to the method that changes the order of the arguments, any existing invocations of that method will still work.

Technically, one should NOT insert keyword arguments anywhere besides the end of the list to guarantee backwards compatibility, but, hey, it happens. For example, you yourself added the new `object_labels_to_filter` argument to `SchemaFilter` BEFORE the existing `prune_empty` argument, which could break anyone's existing code that reads `SchemaFilter(schema, prune_empty)`